include/eve/traits/as_wide.hpp

```diff
@@ -16,7 +16,10 @@
 namespace eve
 {
   template<typename Type, typename Size = expected_cardinal_t<Type>>
-  struct as_wide;
+  struct as_wide
+  {
+    using type = Type;
```

Comment:
This does not seem right. Why do you need this

Reply:
When making a soa_vector of eve::complex, clang tries incorrectly to instantiate as_wide_as<Z,R> in a requires clause of complex += while trying to resolve some `begin() + i` in soa_vector. This instantiation fails and g++ decides the required clause to be false while clang thinks it warrants a hard error.

Reply:
One way to make it less awkward would be to specialize on <T,scalar_cardinal> which is the only case that is not handled specifically. This makes the fix feels less artificial.

Reply:
I don't remember - what's scalar cardinal? Is it fixed<1>?
Can we add requires false to this struct or smth?

Can you add the complex test you mention?

Reply:
scalar_cardinal is a special cardinal different from `fixed<1>` that is returned when asked for the cardinal of a scalar.

Let me add a silly compile test, remove this fix so you can see the problem and we decide on the correct process to fix.